impl/src/main/java/org/jboss/forge/roaster/model/impl/AbstractJavaSourceMemberHolder.java

```diff
@@ -457,6 +459,38 @@ public O implementInterface(Class<?> type)
    public O implementInterface(JavaInterface<?> type)
    {
       O obj = addInterface(type);
+
+      if (type instanceof JavaInterfaceSource) {
+         Set<Import> usedImports = new HashSet<Import>();
+
+         JavaInterfaceSource interfaceSource = (JavaInterfaceSource) type;
+         for (MethodSource<JavaInterfaceSource> method : interfaceSource.getMethods())
+         {
+            if (method.isDefault()) {
+               // Do not add default implementations
+               continue;
+            }
+
+            if (!method.isReturnTypeVoid())
+            {
+               usedImports.add(interfaceSource.getImport(method.getReturnType().getQualifiedName()));
+            }
+
+            for (ParameterSource<JavaInterfaceSource> parameter : method.getParameters())
+            {
+               usedImports.add(interfaceSource.getImport(parameter.getType().getQualifiedName()));
+            }
+         }
+
+         for (Import imprt : interfaceSource.getImports())
+         {
+            if (usedImports.contains(imprt))
+            {
+               addImport(imprt);
```

Comment:
If I understand you correctly, you would like to move the logic of adding imports to the constructor of MethodImpl completely, right?